### PR TITLE
Specs for raw configuration values

### PR DESCRIPTION
**Pull request by robhanlon22**
This is a continuation of #163. This add more spec definitions, specifically around raw, unprocessed configuration, where much of the data can be optionally specified and specified without qualification. With this, a few more functions can have fdefs written for them, and it makes a clearer delineation between input configuration and internal configuration.

### CODE REVIEW

File: src/kaocha/specs.clj [hunk 1]
```diff
@@ -44,10 +77,89 @@
...
+
+(s/def :kaocha.plugin.randomize/randomize?
+  any?)
+
+(s/def :kaocha.plugin.capture-output/capture-output?
+  any?)
+
+(s/def :kaocha/tests (s/coll-of :kaocha/testable))
+
+(s/def :kaocha/testable (s/keys :opt [:kaocha.testable/type
```
Comment (robhanlon22) [inline src/kaocha/specs.clj:89]: I'm wondering if this should be decomposed further into a "test" from config and a "testable" that is the result of load-testables.

Comment (plexus) [reply src/kaocha/specs.clj:89]: Don't we already have that? `:kaocha/testable` vs `:kaocha.test-plan/testable`?

**Comment by plexus** [comment]
Looks good at first glance, sorry for the delay. Could you see what's up with the failing tests?

File: src/kaocha/specs.clj (hunk 1 above)
Comment (robhanlon22) [reply src/kaocha/specs.clj:89]: oh, yes! my bad.

**Comment by plexus** [comment]
@robhanlon22 I took the liberty to push a few commits. The tests were getting stuck, in particular the watch-test, and it was a bit hard to see why. The downside of a test runner testing itself, we're redirecting output all over the place and so sometimes things get hidden that you would rather want to have in your face.

Anyway with some poking around I managed to refine the specs a bit

- bindings can really be anything, we just set these as dynamic bindings, we don't care about the value. Only the key needs to be a valid var name (qualified-symbol?)
- focus and skip are vectors of test-ids, i.e. keywords. Setting it to symbol? broke usage of `--focus`, but setting it to keyword? broke some of our integration tests. It seems so far we have allowed both in tests.edn, I've changed kaocha.config to normalize them to keywords.

Let's see how this fares on CI now. Could you add a changelog entry?

Note that you can run things locally with the CI profile with a simple `CI=true bin/kaocha`.

**Comment by robhanlon22** [comment]
Thanks @plexus! I'll try to get back around to this when things cool down a bit at work :)

**Comment by plexus** [comment]
```
bin/kaocha  --focus 'command-line.print-config/using---print-config'
```

This test is failing now, it needs to be updated to show the new normalized config map.

**Comment by alysbrooks** [comment]
@robhanlon22 I'm looking at older issues/PRs and cleaning them up. Are you interested in picking this up again? If not I might try to wrap this up/see if @plexus is interested in wrapping it up.

**Pull request closed without merging**